util/device_sw_utils/extract_sw_logs.py

```diff
@@ -35,6 +35,21 @@
 RODATA_SECTION = '.rodata'
 
 
+class ArgParseExtendAction(argparse.Action):
+    '''Register the 'extend' action in argparse parser.
+
+    This functionality already exists in Python 3.8+. The minimum supported
+    version for OpenTitan is 3.6.  Hence, we enable this manually for the
+    older versions. The --rodata-sections switch used in this script below
+    needs this to be supported.
+    TODO: Can be removed when OpenTitan mandates Python 3.8 or later.
+    '''
+    def __call__(self, parser, namespace, values, option_string=None):
+        items = getattr(namespace, self.dest) or []
+        items.extend(values)
+        setattr(namespace, self.dest, items)
```

Comment:
This is fine, I guess, but I'm not sure it's really necessary. It might be easier to just concatenate afterwards. So set the action for `--logs-fields-section` to `'append'` and set its default value to `[[RODATA_SECTION]]`. Then flatten the list with something like:
```python3
rodata_sections = list(set(sec for lst in args.rodata_sections for sec in lst))
```
Seems a bit easier to me.

Reply:
Agreed, done.